client/lib/signup/step-actions.js

```diff
@@ -43,7 +43,7 @@ function createCart( callback, dependencies, data ) {
 
 		SignupCart.addToCart( cartKey, [ domainItem ], error => callback( error, providedDependencies ) );
 	} else {
-		createSiteWithCart( callback, dependencies, data );
+		createSiteWithCart( callback, dependencies, data, reduxStore );
 	}
 }
 
```

Comment:
I noticed that we also have `createSiteWithCartAndStartFreeTrial` function that calls `createSiteWithCart` but `reduxStore` param is not passed. I have no idea if that is required, but we should double check if it doesn't break anything.